pycharm ide + seperation of ide pages (#127)
demisto/content-docs: docs/integrations/pycharm-plugin.md

5.  Continue to [Configure Cortex XSOAR on PyCharm](#Cortex XSOAR-setup).

### PyCharm 18.3

1.  From the top menu, click **PyCharm > Preferences**.
2.  Select **Plugins > Marketplace**.
3.  Search for "_Cortex XSOAR_".
4.  When "_Cortex XSOAR Add-on for PyCharm"_ is located, click **Install**.
5.  Continue to [Configure Cortex XSOAR on PyCharm](#Cortex XSOAR-setup).

## Configure Cortex XSOAR on PyCharm

To interact with Cortex XSOAR while working in PyCharm, you need to enter several Cortex XSOAR instance parameters.

1.  Open the project you want to work on in PyCharm. For example, `Cortex XSOAR Content` repository.
2.  In PyCharm, click **Preferences > Tools > Cortex XSOAR Plugin Setup**.
3.  (MacOS users) When prompted for access to your keychain, select **Allow Always**. This will avoid issues later on.
4.  Enter your Cortex XSOAR server URL and port, if necessary.  
    If you are working on a dev instance on localhost, use HTTP (not HTTPS), for example: ([http://localhost:8080](http://localhost:8080/)).
5.  Enter your Cortex XSOAR API key.
6.  Optional: To test the configuration, click **Test**.
7.  Click **Apply**.

> After you successfully configure the plugin several files are automatically downloaded from your Cortex XSOAR instance and saved in your project's root. These files enable you to use the functions you defined in Cortex XSOAR and run the scripts locally in PyCharm.

Reviewer: ```suggestion
After you successfully configure the plugin, several files are automatically downloaded from your Cortex XSOAR instance and saved in your project's root directory. These files enable you to use the functions you defined in Cortex XSOAR and run the scripts locally in PyCharm.
```

*   `CommonServerPython`
*   `CommonServerUserPython`

> *   `Cortex XSOARmock` (enables you to run the scripts locally in PyCharm)

Reviewer 2: ```suggestion
*   `demistomock` (enables you to run the scripts locally in PyCharm)
```

![PyCharm_-_General.png](../doc_imgs/integrations/pycharm-general.png)

## Create a New Cortex XSOAR File

1.  In the IDE project view, select the folder in which to save the new file.
2.  In the top navigation bar, click **File**, and select the file type to create.  
    *   New Cortex XSOAR Automation
    *   New Cortex XSOAR Integration
3.  Enter a name for the file.

New Python and YML files are created. By default, the editor opens the new Python file. You're ready to start writing code.

## Python Files

You can create new Python files, or open existing Python files.

### Create a Python file

> 1.  Under the **Cortex XSOAR Settings** toolbar, you’ll have two button named `Create Cortex XSOAR Automation Configuration` and `Create Cortex XSOAR Integration Configuration`

Reviewer: ```suggestion
1.  Under the **Cortex XSOAR Settings** toolbar, you’ll have two buttons named `Create Cortex XSOAR Automation Configuration` and `Create Cortex XSOAR Integration Configuration`
```

6.  A Python script file would open, and you can edit the code and use `Cortex XSOAR Settings` to edit arguments etc.

### Export a File to Cortex XSOAR

When you run a script in PyCharm, the file is automatically exported to Cortex XSOAR. To manually export a file to Cortex XSOAR, click the **Export to Cortex XSOAR** button in the top actions toolbar.

## Run an Integration or Automation in Cortex XSOAR

From PyCharm, you can run an integration or automation in Cortex XSOAR. When you want to run the script in Cortex XSOAR. The script will be uploaded to, and run in Cortex XSOAR, and the results will be displayed in the `Cortex XSOAR Results` toolbar in PyCharm.

1.  From the **Cortex XSOAR Settings** toolbar, select what to run.  
    *   Run Automation
    *   Run Integration
2.  Enter the necessary arguments.
3.  Click **Export and run in Cortex XSOAR**.

![PyCharm_-_Export_and_Run.png](../doc_imgs/integrations/pycharm-export.png)

## Update Cortex XSOAR Mock Files

In general, you do not need to update Cortex XSOAR mock files. When there is an important update to the mock files, we will announce it in the plugin release notes.

> You might want manually update the mock files after content updates, in which `CommonPythonServer` and `CommonServerUserPython` files are changed or updated. Manually updating the mock files updates the files in your local environment.

Reviewer: ```suggestion
You might want to manually update the mock files after content updates, in which `CommonPythonServer` and `CommonServerUserPython` files are changed or updated. Manually updating the mock files updates the files in your local environment.
```

To update the Cortex XSOAR mock files, select **Tools > Cortex XSOAR Plugin Setup**, and click Update **Cortex XSOAR Mocks**.

![](../doc_imgs/integrations/pycharm-pycharm-mceclip0.png)

## Local Run/Local Debug (Advanced)

Instead of running commands in Cortex XSOAR, you can run them locally in PyCharm. Every demisto action `(demisto.___)` is executed by mock files, and the code runs as a regular Python script.

Since the script is run as a regular Python file, you can debug the code by adding the relevant parameters in the \`demistomock.py\` file, which is located in your project's root. In most cases, you need to edit the return value for several functions.

|Function|Description|Example|
|--- |--- |--- |
|command|Called for `demisto.command()`|return `"my_command_name"`|
|args|Called for `demisto.args()`|return `{ "myArg": "example"}`|

> |parameters|Called fort `demisto.params()`|return `{ "api_key": "example"}`|

Reviewer: ```suggestion
|parameters|Called for `demisto.params()`|return `{ "api_key": "example"}`|
```
